easybuild/easyblocks/generic/pythonpackage.py

```diff
@@ -417,7 +417,7 @@ def determine_install_command(self):
         """
         Determine install command to use.
         """
-        if self.cfg.get('use_pip', False) or self.cfg.get('use_pip_editable', False):
+        if self.cfg.get('use_pip', True) or self.cfg.get('use_pip_editable', False):
```

Comment:
Do we still need this duplication of the default params in EB 5.x or could we just switch to `self.cfg['use_pip']` and fix EasyBlocks not "inheriting" the options this block requires?

Reply:
I think it's wise to keep using `self.cfg.get`, to avoid hard crashes in easyblock to derive from `PythonPackage` but do not inherit the custom easyconfig parameters from `PythonPackage`.
We can make sure the easyblocks we have in the central repository do so, but we don't control all easyblocks out there, and it's a small effort to avoid fallout...

The `self.cfg.get` is a pattern that we should use in all generic easyblocks (and even some custom ones) since they may be used as a base for other easyblocks.

Reply:
I understand that motivation for EB 4.x but I wanted to suggest that such a breakage could be OK for EB 5.x. Not inheriting the parameters has several other implications such as not being able to list available parameters correctly. This introduces some kind of "hidden" parameters and puts burden on upstream EB to be careful to always use the `self.cfg.get` pattern and deal with potential different default values. Technically each instance of those `self.cfg.get` calls would need a check against the extra_parameters section that the default matches and that possibly even across easyblocks (some descendant of e.g. PythonPackage also using `self.cfg.get('use_pip'` would possibly need to be updated too)

Reply:
@Flamefire It's a good point, and I'm happy to follow up on it, but we should do that consistently then, so let's open an issue on this.

The window of opportunity w.r.t. (potentially) breaking changes to include in EasyBuild v5.0 is slowly closing though...

Reply:
Yes, I agree that it is a good point. Let's follow up on this topic in a separate issue or PR. I will merge this PR as is so we can continue testing the `5.0.x` branch for EasyBuild v5.0.